File: TabanDonusumu/TabanDonusumu.java
package tabandonusumu;

import java.util.Scanner;

public class TabanDonusumu {

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.print("Lütfen bir tamsayı giriniz: ");
        int sayi = reader.nextInt();
        if (sayi % 2 == 0) {
            String sonuc = " ";
            while (sayi != 0) {
                int kalan = sayi % 8;
                sayi = sayi / 8;
                sonuc = kalan + sonuc;
            }
            System.out.println("Girilen sayının 8 tabanındaki karşılığı:" + sonuc);
        } else {
            String sonuc = " ";
            while (sayi != 0) {
                int kalan = sayi % 2;
                sayi = sayi / 2;
                sonuc = kalan + sonuc;
            }
            System.out.println("Girilen sayının 2 tabanındaki karşılığı:" + sonuc);
        }
    }
}

File: TabanDonusumu/TabanDonusumu.py
def taban_donusturme():

    sayi = int(input("Bir sayı girin : "))
    donusturulmus_sayi = ''

    if sayi % 2 == 0:
        while sayi > 0:
            donusturulmus_sayi = str(sayi % 2) + donusturulmus_sayi
            sayi = int(sayi / 2)
    else:
        while sayi > 0:
            donusturulmus_sayi = str(sayi % 8) + donusturulmus_sayi
            sayi = int(sayi / 8)

    return donusturulmus_sayi


print(taban_donusturme())